Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -456,26 +475,25 @@ extension ActivityStreamPanel {
             self.presentViewController(controller, animated: true, completion: nil)
         })
 
-        let removeTopSiteAction = ActionOverlayTableViewAction(title: Strings.RemoveFromASContextMenuTitle, iconString: "action_close", handler: { action in
+        let removeTopSiteAction = ActionOverlayTableViewAction(title: Strings.RemoveContextMenuTitle, iconString: "action_remove", handler: { action in
             ASOnyxPing.reportDeleteItemEvent(eventInfo)
             self.hideURLFromTopSites(site.tileURL)
         })
-        
-        let dismissHighlightAction = ActionOverlayTableViewAction(title: Strings.RemoveFromASContextMenuTitle, iconString: "action_close", handler: { action in
+
+        let dismissHighlightAction = ActionOverlayTableViewAction(title: Strings.RemoveContextMenuTitle, iconString: "action_remove", handler: { action in
             ASOnyxPing.reportDeleteItemEvent(eventInfo)
             self.hideFromHighlights(site)
         })
 
-        var actions = [openInNewTabAction, openInNewPrivateTabAction, bookmarkAction, shareAction]
-        switch eventInfo.source {
-            case .highlights: actions.appendContentsOf([dismissHighlightAction, deleteFromHistoryAction])
-            case .topSites: actions.append(removeTopSiteAction)
-            default: break
+        actions.appendContentsOf([bookmarkAction, shareAction])
+
+        switch Section(indexPath.section) {
```

Comment:
Why not just do this in the first switch statement and then add them later? For example, just populate something like an `var extraActions = []` and then update that and append it to actions later?